app/services/punches_service.rb

```diff
@@ -0,0 +1,20 @@
+class PunchesService
+  def self.create(punches, user)
+    punches_date_to_delete = punches.map { |punch| punch['from'] }
+
+    create_delete_punches(punches, punches_date_to_delete, user)
+  end
+
+  private
+
+  def self.create_delete_punches(punches, punches_date_to_delete, user)
+    users_punches = user.punches
+
+    users_punches.transaction do
+      users_punches.by_days(punches_date_to_delete).delete_all
```

Comment:
Is intentional the use of `.delete_all`? `destroy_all` and `delete_all` behave different.

https://spartchou.gitbooks.io/ruby-on-rails-basic/content/syntax/delete_all_vs_destroy_all.html

Reply:
I used the same method that are being used on app/controllers/dashboard_controller.rb